gordo/machine/dataset/base.py

```diff
@@ -230,4 +231,9 @@ def _resample(
             resampled.columns = pd.MultiIndex.from_product(
                 [[series.name], resampled.columns], names=["tag", "aggregation_method"]
             )
-        return resampled.fillna(method="ffill")
+
+        # ensure interpolation max eight hours (8 x 60 minutes) forward
+        # given granularity in minutes
+        granularity = int(re.findall(r"\d+", resolution)[0])
+
+        return resampled.interpolate(limit=int(480 / granularity)).dropna()
```

Comment:
`limit` could be a optinal:
```
         interpolate_limit = None
         if self.interpolate_limit is None:
              interpolate_limit = int(480 / granularity)
         elif self.interpolate_limit != 0:
              interpolate_limit = self.interpolate_limit
         if interpolate_limit is not None:
              return resampled.interpolate(limit=interpolate_limit).dropna()
        else:
              return resampled.fillna(method="ffill")

Reply:
@henrikny I've tried this on several pumps and it seems to be working fine. It actually stops the data from dead tags being forward-filled which was one of our main issues. Very promising!

Of course on certain tags, especially the calculated diff tags, the change in behaviour is very apparent and I've tried to sketch it below. Diff tags in some of the models have a value calculated every 6 hours, before they used to have a step like aspect and now are interpolated. I don't particularly see a big problem with this but would like to hear your opinion.
![ffillvslininterp](https://user-images.githubusercontent.com/35259490/89182107-55875f80-d595-11ea-83d3-ac515c1b9d58.png)

Reply:
Good to hear that it seems to solve some of the issues we've been having with regards to "frozen values".

Regarding the ffill vs. interpolation on calculated diff tags, the actual raw data is also a time-aggregated statistic, probably an average over e.g. 6 hours. In these cases, I think linear interpolation is actually a better method than forward filling. The only situation where this would not be the case, is if we were dealing with nominal scaled data having digital states such as 1=starting, 2=running, 3=shutting down, 4=stopped. As far as I know, we are not dealing with any categorical type of data.